### PR TITLE
refac: Validate wholesale period in before calculating

### PR DESCRIPTION
**Pull request by JesperGomes**
# Description

## Pull-request quality

- [x] The title adheres to [this guide](https://github.com/Mech0z/GitHubGuidelines)
- [x] Tests are written and executed locally
- [ ] Subsystem tests have been tested (by manually deploying to `dev_002` or `sandbox_002`)
- [ ] Documentation has been updated
- [ ] The integration [event catalog](https://energinet.atlassian.net/wiki/spaces/D3/pages/555581556/Event+catalog) has been updated
- [ ] C4 diagrams have been updated and Team Outlaws has been informed about relevant changes

### CODE REVIEW

File: source/databricks/calculation_engine/package/codelists/calculation_type.py [hunk 1]
```diff
@@ -22,3 +22,12 @@ class CalculationType(Enum):
...
     THIRD_CORRECTION_SETTLEMENT = "ThirdCorrectionSettlement"
+
+
+def is_wholesale_calculation_type(calculation_type: CalculationType) -> bool:
+    return (
+        calculation_type == CalculationType.WHOLESALE_FIXING
+        or calculation_type == CalculationType.FIRST_CORRECTION_SETTLEMENT
+        or calculation_type == CalculationType.SECOND_CORRECTION_SETTLEMENT
+        or calculation_type == CalculationType.THIRD_CORRECTION_SETTLEMENT
+    )
```
Comment (BjarkeMeier) [inline source/databricks/calculation_engine/package/codelists/calculation_type.py:33]: We ought to add a unit test that exercises every enum value

File: source/databricks/calculation_engine/tests/infrastructure/test_get_calculator_args.py [hunk 2]
```diff
@@ -188,6 +211,151 @@ def test_raise_system_exit_with_non_zero_code(
...
+                ),
+            ]
+            for calc_type in [
+                CalculationType.AGGREGATION,
+                CalculationType.BALANCE_FIXING,
+                CalculationType.WHOLESALE_FIXING,
+                CalculationType.FIRST_CORRECTION_SETTLEMENT,
+                CalculationType.SECOND_CORRECTION_SETTLEMENT,
+                CalculationType.THIRD_CORRECTION_SETTLEMENT,
+            ]
```
Comment (BjarkeMeier) [inline source/databricks/calculation_engine/tests/infrastructure/test_get_calculator_args.py:334]:
```suggestion
            for calc_type in [t for t in CalculationType]
```

File: source/databricks/calculation_engine/tests/infrastructure/test_get_calculator_args.py [hunk 3]
```diff
@@ -37,6 +37,33 @@ def _get_contract_parameters(filename: str) -> list[str]:
         )
 
 
+def _substitute_period(
+    sys_argv: list[str], period_start_datetime: datetime, period_end_datetime: datetime
+) -> list[str]:
+    for i, item in enumerate(sys_argv):
+        print(item)
```
Comment (BjarkeMeier) [inline source/databricks/calculation_engine/tests/infrastructure/test_get_calculator_args.py:44]:
```suggestion
```

File: source/databricks/calculation_engine/package/calculation/wholesale/subscription_calculators.py [hunk 4]
```diff
@@ -56,45 +54,11 @@ def _calculate_price_per_day(
     prepared_subscriptions: DataFrame,
     calculation_period_start: datetime,
     calculation_period_end: datetime,
-    time_zone: str,
 ) -> DataFrame:
-    days_in_month = _get_days_in_month(
-        calculation_period_start, calculation_period_end, time_zone
-    )
+    days_in_month = (calculation_period_end - calculation_period_start).days
```
[the remark this answers is not in the document]
Comment (JesperGomes) [reply source/databricks/calculation_engine/package/calculation/wholesale/subscription_calculators.py]: Oops. There was a bug here. I have reintroduced time_zone, to ensure we have full days. A test is also added.

**Review by BjarkeMeier — APPROVED** [review]

**Pull request merged**